client/transport/build.gradle

```diff
@@ -55,3 +55,15 @@ namingConventions {
   //we don't have integration tests
   skipIntegTestInDisguise = true
 }
+
+
+javadoc {
+  dependsOn ':core:javadoc'
+  String host;
+  if (version.endsWith("-SNAPSHOT")) {
+    host = "https://snapshots.elastic.co";
+  } else {
+    host = "https://artifacts.elastic.co";
+  }
+  options.linksOffline host + "/javadoc/org/elasticsearch/elasticsearch/${version}", "../../core/build/docs/javadoc/"
```

Comment:
it is appealing to do it generically, yet I think things will quickly become complicated. For instance the order of the linksOffline options (rest and core) in rest-high-level matters, as we want to do first `org.elasticsearch.client.*` and then `org.elasticsearch.*`, otherwise the links to the RestClient classes point to core instead (non existing html files). I think doing it only for elasticsearch deps may work, but given that these are the only projects that we promote the javadocs for at the moment (links are in the docs pages), I don't see a big advantage in doing it generically at the moment.